src/awkward/_nplikes/__init__.py

```diff
@@ -73,6 +75,10 @@ def nplike_of(*arrays, default: D = _UNSET) -> NumpyLike | D:
                 if cls.is_own_array(array):
                     nplikes.add(cls.instance())
                     break
+            else:
+                # Explicitly test non-layout ArrayBuilder types
+                if isinstance(array, (_ext.ArrayBuilder, ak.highlevel.ArrayBuilder)):
+                    nplikes.add(Numpy.instance())
```

Comment:
I don't love that we need to import layout-specific types here. I'd prefer to do this via e.g. attribute inspection. However, the proper way would be to define a `backend` property on the high & lowlevel array-builder types, which would expose `backend` to the L1 API.

Reply:
ArrayBuilder is not a layout, and anything that an ArrayBuilder builds is naturally on the CPU, ready for `Numpy.instance()`. This doesn't seem like it's overreaching to me.